Option in category constructor for avoiding any overhead

sebastianpos opened the pull request:
The idea is to provide an option in the category constructor that reduces CAP's overhead to zero, e.g., when the option is enabled, the call of `KernelLift` directly triggers GAP's method selection in order to search for the function that was given by the user (or by the derivation machinery).

zickgraf commented:
In PR #378 I'm also touching `CapInternalInstallAdd`. Since your change seems to be "wrap the existing code in an `if` and add an `else` case" I think it would be much simpler to rebase this PR on #378 than the other way round. So, could you maybe merge #378 first?

sebastianpos commented:
Okay.

sebastianpos commented:
I rebased this PR to master (which was unfortunately harder than I thought).
I also changed the indentation such that the diff is now easier to read.
